File: OTEAServer/Models/SubAmbit.cs
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations;

namespace OTEAServer.Models
{
    /// <summary>
    /// Model class for first level division of ambits
    /// Author: Pablo Ahíta del Barrio
    /// Version: 1
    /// </summary>
    public class SubAmbit
    {
        /// <summary>
        /// Class constructor
        /// </summary>
        /// <param name="idSubAmbit">First division of ambit identifier</param>
        /// <param name="idAmbit">Ambit identifier</param>
        /// <param name="descriptionEnglish">SubAmbit description in English</param>
        /// <param name="descriptionSpanish">SubAmbit description in Spanish</param>
        /// <param name="descriptionFrench">SubAmbit description in French</param>
        /// <param name="descriptionBasque">SubAmbit description in Basque</param>
        /// <param name="descriptionCatalan">SubAmbit description in Catalan</param>
        /// <param name="descriptionDutch">SubAmbit description in Dutch</param>
        /// <param name="descriptionGalician">SubAmbit description in Galician</param>
        /// <param name="descriptionGerman">SubAmbit description in German</param>
        /// <param name="descriptionItalian">SubAmbit description in Italian</param>
        /// <param name="descriptionPortuguese">SubAmbit description in Portuguese</param>
        public SubAmbit(int idSubAmbit, int idAmbit, String descriptionEnglish, String descriptionSpanish, string descriptionFrench, string descriptionBasque, string descriptionCatalan, string descriptionDutch, string descriptionGalician, string descriptionGerman, string descriptionItalian, string descriptionPortuguese)
        {
            this.idSubAmbit = idSubAmbit;
            this.idAmbit = idAmbit;
            this.descriptionSpanish = descriptionSpanish;
            this.descriptionEnglish = descriptionEnglish;
            this.descriptionFrench = descriptionFrench;
            this.descriptionBasque = descriptionBasque;
            this.descriptionCatalan = descriptionCatalan;
            this.descriptionDutch = descriptionDutch;
            this.descriptionGalician = descriptionGalician;
            this.descriptionGerman = descriptionGerman;
            this.descriptionItalian = descriptionItalian;
            this.descriptionPortuguese = descriptionPortuguese;
        }

        /// <summary>
        /// First division of ambit identifier
        /// </summary>
        [JsonPropertyName("idSubAmbit")]
        public int idSubAmbit { get; set; }

        /// <summary>
        /// Ambit identifier
        /// </summary>
        [Key]
        [JsonPropertyName("idAmbit")]
        public int idAmbit { get; set; }

        /// <summary>
        /// SubAmbit description in Spanish
        /// </summary>
        [JsonPropertyName("descriptionSpanish")]
        public string descriptionSpanish { get; set; }

        /// <summary>
        /// SubAmbit description in English
        /// </summary>
        [JsonPropertyName("descriptionEnglish")]
        public string descriptionEnglish { get; set; }

        /// <summary>
        /// SubAmbit description in French
        /// </summary>
        [JsonPropertyName("descriptionFrench")]
        public string descriptionFrench { get; set; }

        /// <summary>
        /// SubAmbit description in Basque
        /// </summary>
        [JsonPropertyName("descriptionBasque")]
        public string descriptionBasque { get; set; }

        /// <summary>
        /// SubAmbit description in Catalan
        /// </summary>
        [JsonPropertyName("descriptionCatalan")]
        public string descriptionCatalan { get; set; }

        /// <summary>
        /// SubAmbit description in Dutch
        /// </summary>
        [JsonPropertyName("descriptionDutch")]
        public string descriptionDutch { get; set; }


        /// <summary>
        /// SubAmbit description in Galician
        /// </summary>
        [JsonPropertyName("descriptionGalician")]
        public string descriptionGalician { get; set; }

        /// <summary>
        /// SubAmbit description in German
        /// </summary>
        [JsonPropertyName("descriptionGerman")]
        public string descriptionGerman { get; set; }

        /// <summary>
        /// SubAmbit description in Italian
        /// </summary>
        [JsonPropertyName("descriptionItalian")]
        public string descriptionItalian { get; set; }

        /// <summary>
        /// SubAmbit description in Portuguese
        /// </summary>
        [JsonPropertyName("descriptionPortuguese")]
        public string descriptionPortuguese { get; set; }
    }
}

File: connection/src/main/java/cli/indicators/SubAmbit.java
package cli.indicators;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Model class for first level division of ambits
 *
 * @author Pablo Ahíta del Barrio
 * @version 1
 * */
public class SubAmbit implements Serializable {

    /**First level division of ambit identifier*/
    @SerializedName("idSubAmbit")
    public int idSubAmbit;

    /**Ambit identifier*/
    @SerializedName("idAmbit")
    public int idAmbit;

    /**SubAmbit  description in English*/
    @SerializedName("descriptionEnglish")
    public String descriptionEnglish;

    /**SubAmbit  description in Spanish*/
    @SerializedName("descriptionSpanish")
    public String descriptionSpanish;

    /**SubAmbit  description in French*/
    @SerializedName("descriptionFrench")
    public String descriptionFrench;

    /**SubAmbit  description in Basque*/
    @SerializedName("descriptionBasque")
    public String descriptionBasque;

    /**SubAmbit  description in Catalan*/
    @SerializedName("descriptionCatalan")
    public String descriptionCatalan;

    /**SubAmbit  description in Dutch*/
    @SerializedName("descriptionDutch")
    public String descriptionDutch;

    /**SubAmbit  description in Galician*/
    @SerializedName("descriptionGalician")
    public String descriptionGalician;

    /**SubAmbit  description in German*/
    @SerializedName("descriptionGerman")
    public String descriptionGerman;

    /**SubAmbit  description in Italian*/
    @SerializedName("descriptionItalian")
    public String descriptionItalian;

    /**SubAmbit  description in Portuguese*/
    @SerializedName("descriptionPortuguese")
    public String descriptionPortuguese;

    /**
     * Class constructor
     *
     * @param idSubAmbit - SubAmbit identifier
     * @param idAmbit - Ambit identifier
     * @param descriptionEnglish - SubAmbit  description in English
     * @param descriptionSpanish - SubAmbit  description in Spanish
     * @param descriptionFrench - SubAmbit  description in French
     * @param descriptionBasque - SubAmbit  description in Basque
     * @param descriptionCatalan - SubAmbit  description in Catalan
     * @param descriptionDutch - SubAmbit  description in Dutch
     * @param descriptionGalician - SubAmbit  description in Galician
     * @param descriptionGerman - SubAmbit  description in German
     * @param descriptionItalian - SubAmbit  description in Italian
     * @param descriptionPortuguese - SubAmbit  description in Portuguese
     * */
    public SubAmbit(int idSubAmbit, int idAmbit, String descriptionEnglish, String descriptionSpanish, String descriptionFrench,String descriptionBasque, String descriptionCatalan, String descriptionDutch,String descriptionGalician, String descriptionGerman, String descriptionItalian,String descriptionPortuguese){
        setIdSubAmbit(idSubAmbit);
        setIdAmbit(idAmbit);
        setDescriptionEnglish(descriptionEnglish);
        setDescriptionSpanish(descriptionSpanish);
        setDescriptionFrench(descriptionFrench);
        setDescriptionBasque(descriptionBasque);
        setDescriptionCatalan(descriptionCatalan);
        setDescriptionDutch(descriptionDutch);
        setDescriptionGalician(descriptionGalician);
        setDescriptionGerman(descriptionGerman);
        setDescriptionItalian(descriptionItalian);
        setDescriptionPortuguese(descriptionPortuguese);
    }

    /**
     * Method that obtains the subAmbit identifier
     *
     * @return SubAmbit identifier
     * */
    public int getIdSubAmbit() {
        return idSubAmbit;
    }

    /**
     * Method that sets the new subAmbit identifier
     *
     * @param idSubAmbit - SubAmbit identifier
     * */
    public void setIdSubAmbit(int idSubAmbit) {
        this.idSubAmbit = idSubAmbit;
    }

    /**
     * Method that obtains the ambit identifier
     *
     * @return Ambit identifier
     * */
    public int getIdAmbit() {
        return idAmbit;
    }

    /**
     * Method that sets the new ambit identifier
     *
     * @param idAmbit - Ambit identifier
     * */
    public void setIdAmbit(int idAmbit) {
        this.idAmbit = idAmbit;
    }

    /**
     * Method that obtains the subSubAmbit description in English
     *
     * @return Ambit description in English
     * */
    public String getDescriptionEnglish() {
        return descriptionEnglish;
    }

    /**
     * Method that sets the new subSubAmbit description in Portuguese
     *
     * @param descriptionEnglish - Ambit description in English
     * */
    public void setDescriptionEnglish(String descriptionEnglish) {
        this.descriptionEnglish = descriptionEnglish;
    }

    /**
     * Method that obtains the subSubAmbit description in Spanish
     *
     * @return Ambit description in Spanish
     * */
    public String getDescriptionSpanish() {
        return descriptionSpanish;
    }

    /**
     * Method that sets the new subSubAmbit description in Spanish
     *
     * @param descriptionSpanish - Ambit description in Spanish
     * */
    public void setDescriptionSpanish(String descriptionSpanish) {
        this.descriptionSpanish = descriptionSpanish;
    }

    /**
     * Method that obtains the subSubAmbit description in French
     *
     * @return Ambit description in French
     * */
    public String getDescriptionFrench() {
        return descriptionFrench;
    }

    /**
     * Method that sets the new subSubAmbit description in Portuguese
     *
     * @param descriptionFrench - Ambit description in French
     * */
    public void setDescriptionFrench(String descriptionFrench) {
        this.descriptionFrench = descriptionFrench;
    }

    /**
     * Method that obtains the subSubAmbit description in Basque
     *
     * @return Ambit description in Basque
     * */
    public String getDescriptionBasque() {
        return descriptionBasque;
    }

    /**
     * Method that sets the new subSubAmbit description in Basque
     *
     * @param descriptionBasque - Ambit description in Basque
     * */
    public void setDescriptionBasque(String descriptionBasque) {
        this.descriptionBasque = descriptionBasque;
    }

    /**
     * Method that obtains the subSubAmbit description in Catalan
     *
     * @return Ambit description in Catalan
     * */
    public String getDescriptionCatalan() {
        return descriptionCatalan;
    }

    /**
     * Method that sets the new subSubAmbit description in Catalan
     *
     * @param descriptionCatalan - Ambit description in Catalan
     * */
    public void setDescriptionCatalan(String descriptionCatalan) {
        this.descriptionCatalan = descriptionCatalan;
    }

    /**
     * Method that obtains the subSubAmbit description in Dutch
     *
     * @return Ambit description in Dutch
     * */
    public String getDescriptionDutch() {
        return descriptionDutch;
    }

    /**
     * Method that sets the new subSubAmbit description in Dutch
     *
     * @param descriptionDutch - Ambit description in Dutch
     * */
    public void setDescriptionDutch(String descriptionDutch) {
        this.descriptionDutch = descriptionDutch;
    }

    /**
     * Method that obtains the subSubAmbit description in Galician
     *
     * @return Ambit description in Galician
     * */
    public String getDescriptionGalician() {
        return descriptionGalician;
    }

    /**
     * Method that sets the new subSubAmbit description in Galician
     *
     * @param descriptionGalician - Ambit description in Galician
     * */
    public void setDescriptionGalician(String descriptionGalician) {
        this.descriptionGalician = descriptionGalician;
    }

    /**
     * Method that obtains the subSubAmbit description in German
     *
     * @return Ambit description in German
     * */
    public String getDescriptionGerman() {
        return descriptionGerman;
    }

    /**
     * Method that sets the new subSubAmbit description in German
     *
     * @param descriptionGerman - Ambit description in German
     * */
    public void setDescriptionGerman(String descriptionGerman) {
        this.descriptionGerman = descriptionGerman;
    }

    /**
     * Method that obtains the subSubAmbit description in Italian
     *
     * @return Ambit description in Italian
     * */
    public String getDescriptionItalian() {
        return descriptionItalian;
    }

    /**
     * Method that sets the new subSubAmbit description in Italian
     *
     * @param descriptionItalian - Ambit description in Italian
     * */
    public void setDescriptionItalian(String descriptionItalian) {
        this.descriptionItalian = descriptionItalian;
    }

    /**
     * Method that obtains the subSubAmbit description in Portuguese
     *
     * @return Ambit description in Portuguese
     * */
    public String getDescriptionPortuguese() {
        return descriptionPortuguese;
    }

    /**
     * Method that sets the new subSubAmbit description in Portuguese
     *
     * @param descriptionPortuguese - Ambit description in Portuguese
     * */
    public void setDescriptionPortuguese(String descriptionPortuguese) {
        this.descriptionPortuguese = descriptionPortuguese;
    }
}
